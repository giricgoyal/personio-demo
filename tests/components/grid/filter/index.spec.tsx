import { fireEvent } from '@testing-library/react'
import React from 'react'
import Filter from 'src/components/grid/filter'
import { render } from 'tests/test-utils'

describe('components/grid/filter', () => {
    let container, getByTestId, onChangeHandler, rerender
    const filterOptions = [
        {
            label: 'col1',
            value: 'col1',
            filterValues: ['val1', 'val2', 'val3'],
        },
        {
            label: 'col2',
            value: 'col2',
            filterValues: ['1', '2', '3'],
        },
    ]

    beforeEach(() => {
        onChangeHandler = jest.fn()
        ;({ container, getByTestId, rerender } = render(
            <Filter filterOptions={filterOptions} onChange={onChangeHandler} filterBy="" />,
        ))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('When filterBy is provided', () => {
        beforeEach(() => {
            rerender(<Filter filterOptions={filterOptions} onChange={onChangeHandler} filterBy="col1:val2" />)
        })

        test('should render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })

        describe('onChange', () => {
            describe('when filter value is changed', () => {
                beforeEach(() => {
                    fireEvent.change(getByTestId('filter-value-input'), {
                        target: {
                            value: 'val1',
                        },
                    })
                })

                test('should call the onChange handler', () => {
                    expect(onChangeHandler).toHaveBeenCalledWith('col1:val1')
                })
            })
        })
    })

    describe('onChange', () => {
        describe('When filter is changed', () => {
            beforeEach(() => {
                fireEvent.change(getByTestId('filterby-select'), {
                    target: {
                        value: 'col2',
                    },
                })
            })

            test('should call the onChange handler', () => {
                expect(onChangeHandler).toHaveBeenCalledWith('col2:')
            })
        })

        describe('when filter value is changed', () => {
            beforeEach(() => {
                fireEvent.change(getByTestId('filter-value-input'), {
                    target: {
                        value: '2',
                    },
                })
            })

            test('should call the onChange handler', () => {
                expect(onChangeHandler).toHaveBeenCalledWith('')
            })
        })
    })
})
