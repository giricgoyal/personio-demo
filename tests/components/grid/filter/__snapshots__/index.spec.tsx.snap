// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid/filter When filterBy is provided should render and match snapshot 1`] = `
<div>
  <div>
    <i
      class="fas fa-filter "
    />
    <label
      class="select"
    >
      <span
        class="select__label"
      />
      <select
        class="select__element"
        data-testid="filterby-select"
      >
        <option
          selected=""
          value="col1"
        >
          col1
        </option>
        <option
          value="col2"
        >
          col2
        </option>
      </select>
    </label>
    <label
      class="combo-box"
    >
      <input
        data-testid="filter-value-input"
        list="filter-value-input"
        value="val2"
      />
      <button
        class="button button--tertiary"
        data-testid="combobox-value-clear"
      >
        <i
          class="fas fa-times "
        />
      </button>
    </label>
    <datalist
      id="filter-value-input"
    >
      <option
        value="val1"
      />
      <option
        value="val2"
      />
      <option
        value="val3"
      />
    </datalist>
  </div>
</div>
`;

exports[`components/grid/filter should render and match snapshot 1`] = `
<div>
  <div>
    <i
      class="fas fa-filter "
    />
    <label
      class="select"
    >
      <span
        class="select__label"
      />
      <select
        class="select__element"
        data-testid="filterby-select"
      >
        <option
          value="col1"
        >
          col1
        </option>
        <option
          value="col2"
        >
          col2
        </option>
      </select>
    </label>
    <label
      class="combo-box"
    >
      <input
        data-testid="filter-value-input"
        list="filter-value-input"
        value=""
      />
    </label>
  </div>
</div>
`;
